ts/commitments.ts

```diff
@@ -225,58 +226,70 @@ export class MassMigrationCommitment extends Commitment {
             }
         };
     }
-    public toBatch() {
-        return new MassMigrationBatch([this]);
+    public async toBatch() {
+        return await MassMigrationBatch.new([this]);
     }
 }
 
 export class Create2TransferCommitment extends TransferCommitment {
-    public toBatch(): Create2TransferBatch {
-        return new Create2TransferBatch([this]);
+    public async toBatch(): Promise<Create2TransferBatch> {
+        return await Create2TransferBatch.new([this]);
     }
 }
 
 export class Batch {
-    private tree: MemoryTree;
-    constructor(public readonly commitments: Commitment[]) {
-        this.tree = MemoryTree.merklize(commitments.map(c => c.hash()));
+    public tree: MemoryTree = MemoryTree.new(1);
+    constructor(public readonly commitments: Commitment[]) {}
+
+    static async new(commitments: Commitment[]) {
+        const batch = new Batch(commitments);
+        batch.tree = await MemoryTree.merklize(commitments.map(c => c.hash()));
```

Comment:
creates a Batch object with default MemoryTree and then sets a new Tree to it after init